Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/IPay88TokenizationViewModel.swift

```diff
@@ -0,0 +1,547 @@
+//
+//  IPay88TokenizationViewModel.swift
+//  PrimerSDK
+//
+//  Created by Evangelos on 12/12/22.
+//
+
+#if canImport(UIKit)
+
+import Foundation
+import UIKit
+
+#if canImport(PrimerIPay88SDK)
+import PrimerIPay88SDK
+#endif
+
+class IPay88TokenizationViewModel: PaymentMethodTokenizationViewModel {
+    
+    deinit {
+        log(logLevel: .debug, message: "🧨 deinit: \(self.self) \(Unmanaged.passUnretained(self).toOpaque())")
+    }
+    
+#if canImport(PrimerIPay88SDK)
+    private var backendCallbackUrl: URL!
+    private var primerTransactionId: String!
+    private var statusUrl: URL!
+    private var resumeToken: String!
+    private var primerIPay88ViewController: PrimerIPay88ViewController!
+    private var primerIPay88Payment: PrimerIPay88Payment!
+    private var didCancel: (() -> Void)?
+    private var didComplete: (() -> Void)?
+    private var didFail: ((_ err: PrimerError) -> Void)?
+#endif
+    
+    private lazy var iPay88NumberFormatter: NumberFormatter = {
+        let numberFormatter = NumberFormatter()
+        numberFormatter.numberStyle = .decimal
+        numberFormatter.decimalSeparator = "."
+        numberFormatter.groupingSeparator = ","
+        numberFormatter.maximumFractionDigits = 2
+        numberFormatter.minimumFractionDigits = 2
+        return numberFormatter
+    }()
+    
+    override func validate() throws {
+        guard let decodedJWTToken = PrimerAPIConfigurationModule.decodedJWTToken, decodedJWTToken.isValid else {
+            let err = PrimerError.invalidClientToken(
+                userInfo: ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"],
```

Comment:
I'm seeing you using this everywhere. I understand you cannot use another function cuz `#file` and `#function` would point to another place. 
But could use preprocessing?

```
#define ANALYTICS_DEFAULT_USER_INFO ["file": #file, "class": "\(Self.self)", "function": #function, "line": "\(#line)"]
```

Reply:
+1